src/proto/Cosmos.proto

```diff
@@ -85,6 +85,53 @@ message Message {
         string type_prefix = 3;
     }
 
+    // wasm/MsgExecuteContract
+    message WasmExecuteContract {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        // base64-encoded payload of contract execution message (refer to below)
+        string execute_msg = 3;
+
+        // used in case you are sending native tokens along with this message
+        repeated string coin = 4;
+    }
+
+    // transfer within wasm/MsgExecuteContract
+    message WasmExecuteContractTransfer {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        int64 amount = 3;
+
+        string recipient_address = 4;
+    }
+
+    // send within wasm/MsgExecuteContract
+    message WasmExecuteContractSend {
+        // sender address
+        string sender_address = 1;
+
+        // token contract address
+        string contract_address = 2;
+
+        int64 amount = 3;
```

Comment:
```suggestion
        int64 amount = 3;
```

is int64 enough for the amount?

Reply:
it's actually 128 bits in wasm impl., I change.